src/urdf/util.cc

```diff
@@ -201,32 +202,29 @@ namespace hpp {
         {
           if (!urdfTree)
             throw std::invalid_argument ("Failed to parse URDF. Use check_urdf command to know what's wrong.");
-          if (baseJoint != 0)
-            throw std::invalid_argument ("Only appending robots at the world is supported.");
 
-          Model& model = robot->model();
-          const JointIndex idFirstJoint = model.joints.size();
-          const FrameIndex idFirstFrame = model.frames.size();
+          ModelPtr_t model = (baseFrame==0 ? robot->modelPtr() : ModelPtr_t(new Model));
+          const JointIndex idFirstJoint = model->joints.size();
+          const FrameIndex idFirstFrame = model->frames.size();
           if (rootType == "anchor")
-            ::pinocchio::urdf::buildModel(urdfTree, model, verbose);
+            ::pinocchio::urdf::buildModel(urdfTree, *model, verbose);
           else
-            ::pinocchio::urdf::buildModel(urdfTree, buildJoint(rootType), model, verbose);
-          robot->createData();
+            ::pinocchio::urdf::buildModel(urdfTree, buildJoint(rootType), *model, verbose);
 
           hppDout (notice, "Finished parsing URDF file.");
 
           GeomModel geomModel;
 
           std::vector<std::string> baseDirs = ::pinocchio::rosPaths();
-          fcl::MeshLoaderPtr loader (new fcl::CachedMeshLoader);
-          ::pinocchio::urdf::buildGeom(model, urdfStream, ::pinocchio::COLLISION, geomModel, baseDirs, loader);
+          static fcl::MeshLoaderPtr loader (new fcl::CachedMeshLoader (fcl::BV_OBB));
```

Comment:
@stonneau your problem comes from here. Before, we used BV_OBBRSS. OBB-RSS bounding volume allows distance computation, while OBB does not. I switched to OBB because it takes much less memory.

 I see two ways of solving this issue:
- modify hpp-fcl so that OBB can handle distance computation. I believe that it isn't hard to implement but that it will be slower than OBBRSS.
- make the BV type modifiable somehow.

Reply:
The next HPP release will be done with OBBRSS. We will include the implementation of the features above in a next release.

Reply:
Adressed in https://github.com/humanoid-path-planner/hpp-fcl/pull/70

Reply:
I haven't benchmarked the performances though. @stonneau are they critical for its use in rbprm ?